### PR TITLE
change join table from allergen to allergy

### PR DESCRIPTION
**Pull request by telegraham**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,11 +80,11 @@ should return the ingredient instance
 - `RecipeIngredient#recipe`
 should return the recipe instance
 
-### `Allergen`
-An Allergen is a join between a user and an ingredient.  This is a has-many-through relationship.  What methods should an instance of this model respond to?
+### `Allergy`
```
Comment (sgharms) [inline README.md]: This subtlety below comes AFTER it's domain model impact is needed in `declare_allergy`. Move this bloc before `User`?

File: README.md [hunk 2]
```diff
@@ -34,8 +34,8 @@ should return all of the user instances
 should return all of the recipes this user has recipe cards for
 - `User#add_recipe_card`
 should accept a recipe instance as an argument, as well as a date and rating, and create a new recipe card for this user and the given recipe
-- `User#declare_allergen`
-should accept an ingredient instance as an argument, and create a new allergen instance for this user and the given ingredient
+- `User#declare_allergy`
+should accept an ingredient instance as an argument, and create a new allergy instance for this user and the given ingredient
```
Comment (sgharms) [inline README.md]: Formatting...

File: README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ should return the user instances who have recipe cards with this recipe
 - `Recipe#ingredients`
 should return all of the ingredients in this recipe
 - `Recipe#allergens`
-should return all of the ingredients in this recipe that are allergens
+should return all of the ingredients in this recipe that users are allergic to
```
Comment (sgharms) [inline README.md]: "...`Ingredient`s in this recipe that are allergens for `User`s in our system"

This hints that the join is the key of Allergen to User via Allergy.

**Review by sgharms — DISMISSED** [review]
There's definitely some subtlety that was lost in this body that this PR makes better.

## Specific Issues

* The formatting in here is not helpful for inferring what's up. `ClassNames` should be ticked to make it pop for when we're discussing instances or not. I had to re-parse some of these changes. Could you please format the ClassNames in a commit?
* I did some word suggestion changes that I think make the logic flow better
* Suggested some ordering ideas

File: README.md (hunk 3 above)
Comment (timothylevi) [inline README.md]:
```suggestion
should return all of the `Ingredient`s in this recipe that are allergens for `User`s in our system.
```

File: README.md (hunk 2 above)
Comment (timothylevi) [inline README.md]:
```suggestion
should accept an`Ingredient` instance as an argument, and create a new `Allergy` instance for this `User` and the given `Ingredient`
```

**Pull request merged**